### PR TITLE
Sort: Stable sort, ignore non-printing, last resort comparison.

### PR DESCRIPTION
**Pull request by electricboogie**
Implemented stable sort, ignore non-printing, last resort comparison.

Last resort comparison fixes all sorts of ordering issues.

Created a general numeric compare option and made numeric compare a stub pointing to that.

Switched to FNV hash for random shuffle because it has better small input characteristics.

Removed my tests within sort.rs.

Created new tests in test_sort.rs.

### CODE REVIEW

File: src/uu/sort/src/sort.rs [hunk 1]
```diff
@@ -487,24 +514,39 @@ fn get_leading_number(a: &str) -> &str {
...
+fn leading_num_numlines(a: &str) -> &str {
+    // Trim and remove any leading zeros
+    let s = a.trim().trim_start_matches('0');
+
+    // Get first char
+    let c = s.chars().nth(0).unwrap_or('\0');
+
+    // Empty, non-number lines, whitespace and are treated as same for dedup
+    if s.is_empty() {
+        return "";
```
Comment (ArniDagur) [inline src/uu/sort/src/sort.rs:529]: I'd get rid of the `return` statements here and let the value of the if statement be returned.

Comment (electricboogie) [reply src/uu/sort/src/sort.rs:529]: Thanks for reviewing.  Absolutely.

**Comment by sylvestre** [comment]
@electricboogie you don't want to finish this?

**Comment by electricboogie** [comment]
Still want to finish.   I want to figure out how to squash these (too many!) commits and remove the .DS_Store files (via .gitignore?) from the commit.   I'll resubmit, if that's alright, soon.

**Pull request closed without merging**